src/gax-internal/src/http.rs

```diff
@@ -451,4 +484,45 @@ mod tests {
         assert_eq!(info.client_artifact, "test-artifact");
         assert_eq!(info.default_host, "test.googleapis.com");
     }
+
+    #[test_case(None, "test.googleapis.com"; "default")]
+    #[test_case(Some("http://www.googleapis.com"), "test.googleapis.com"; "global")]
+    #[test_case(Some("http://private.googleapis.com"), "test.googleapis.com"; "VPC-SC private")]
+    #[test_case(Some("http://restricted.googleapis.com"), "test.googleapis.com"; "VPC-SC restricted")]
+    #[test_case(Some("http://test-my-private-ep.p.googleapis.com"), "test.googleapis.com"; "PSC custom endpoint")]
+    #[test_case(Some("https://us-central1-test.googleapis.com"), "us-central1-test.googleapis.com"; "locational endpoint")]
+    #[test_case(Some("https://test.us-central1.rep.googleapis.com"), "test.us-central1.rep.googleapis.com"; "regional endpoint")]
+    #[test_case(Some("https://test.my-universe-domain.com"), "test.googleapis.com"; "universe domain")]
+    #[test_case(Some("localhost:5678"), "test.googleapis.com"; "emulator")]
```

Comment:
From go/rep-mtls, but I am not sure if these are finalized:

* `content-{service}.{region}.rep.mtls.googleapis.com`
* `{service}.{region}.rep.mtls.googleapis.com`

Reply:
Ack. As it stands we will use the default host for these, which might just work. (In my testing, both the default host and the regional host worked for regional endpoints).